schema_based_on_diagram.sql

```diff
@@ -0,0 +1,41 @@
+CREATE TABLE patients (
+    id INT GENERATED ALWAYS AS IDENTITY,
+    name varchar,
+    date_of_birth date,
+    PRIMARY KEY (id)
+);
+CREATE TABLE treatments (
+    id INT GENERATED ALWAYS AS IDENTITY,
+    type varchar,
+    name varchar,
+    PRIMARY KEY (id)
+);
+CREATE TABLE medical_histories (
+    id INT GENERATED ALWAYS AS IDENTITY,
+    admitted_at TIMESTAMP,
+    patient_id INT,
+    status varchar,
+    FOREIGN KEY (patient_id) REFERENCES  patients(id),
+    FOREIGN KEY (id) REFERENCES treatments(id),
+    PRIMARY KEY (id)
+);
+CREATE TABLE invoices (
+    id INT GENERATED ALWAYS AS IDENTITY,
+    total_amount DECIMAL,
+    generated_at TIMESTAMP,
+    payed_at TIMESTAMP,
+    medical_history_id INT,
+    FOREIGN KEY (medical_history_id) REFERENCES  medical_histories(id),
+    PRIMARY KEY (id)
+);
+CREATE TABLE invoice_items (
+    id INT GENERATED ALWAYS AS IDENTITY,
+    unit_price DECIMAL,
+    quantity INT,
+    total_price DECIMAL,
+    invoice_id INT,
+    treatment_id INT,
+    FOREIGN KEY (invoice_id) REFERENCES  invoices(id),
+    FOREIGN KEY (treatment_id) REFERENCES  treatments(id),
+    PRIMARY KEY (id)
```

Comment:
- [ ] Please make sure that all FOREIGN KEYS have indexes created.
![image](https://user-images.githubusercontent.com/83803414/197849806-85039100-34ce-48fa-a08e-aa81db32c526.png)